R/tm_missing_data.R

```diff
@@ -315,75 +316,69 @@ encoding_missing_data <- function(id, summary_per_patient = FALSE, ggtheme, data
       )
     ),
     hr(),
-    teal::get_rcode_ui(ns("rcode"))
+    teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code")
   )
 }
 
 #' @importFrom rlang .data
-srv_missing_data <- function(id, datasets, reporter, dataname, plot_height, plot_width, ggplot2_args) {
+srv_missing_data <- function(id, data, reporter, filter_panel_api, dataname, plot_height, plot_width, ggplot2_args) {
   with_reporter <- !missing(reporter) && inherits(reporter, "Reporter")
+  with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelApi")
   moduleServer(id, function(input, output, session) {
-    teal.code::init_chunks()
-
     prev_group_by_var <- reactiveVal("")
 
-    raw_data <- reactive({
-      # unfiltered data maintains all attributes such as labels
-      datasets$get_data(dataname, filtered = FALSE)
-    })
+    data_r <- data[[dataname]]
 
-    data <- reactive({
-      datasets$get_data(dataname, filtered = TRUE)
-    })
-
-    data_keys <- reactive({
-      datasets$get_keys(dataname)
+    data_keys <- reactive(attr(data, "join_keys")$get(dataname)[[dataname]])
+    data_parent_keys <- reactive({
+      keys <- attr(data, "join_keys")$get(dataname)
+      if ("ADSL" %in% names(keys)) {
+        keys[["ADSL"]]
+      } else {
+        keys[[dataname]]
+      }
     })
 
     # chunks needed by all three outputs stored here
-    common_code_chunks <- reactive({
-      # Create a private stack for this function only.
-      common_stack <- teal.code::chunks$new()
-      common_stack_push <- function(...) {
-        teal.code::chunks_push(..., chunks = common_stack)
-      }
-
+    common_code_q <- reactive({
       anl_name <- dataname
-      anl <- datasets$get_data(dataname, filtered = TRUE)
+      anl <- data_r()
       assign(anl_name, anl)
-      teal.code::chunks_reset(chunks = common_stack)
       group_var <- input$group_by_var
 
-      if (!is.null(selected_vars()) && length(selected_vars()) != ncol(anl)) {
-        common_stack_push(
-          id = "ANL_selection_call",
-          expression = substitute(
+      quosure <- teal.code::new_quosure(data)
```

Comment:
As discussed with @gogonzo we should make the quosure object only have data()[[dataname]] inside (with a isolate for the code)